tsdb/index.go

```diff
@@ -2001,7 +2008,7 @@ func (is IndexSet) MeasurementSeriesKeyByExprIterator(name []byte, expr influxql
 	}
 	if ids == nil {
 		release()
-		return nil, nil
+		release = nil
```

Comment:
For us to successfully use a nil `measurementSeriesKeyByExprIterator`, we would have to return a boxed nil, which is ugly.

Instead just set the `ids` to nil and the methods can check that.